plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/rest/TrinoRESTCatalog.java

```
@@ -0,0 +1,364 @@
+        if (!sessionCatalog.dropTable(convert(session), toIdentifier(schemaTableName))) {
+            throw new TrinoException(ICEBERG_CATALOG_ERROR, format("Failed to drop table: %s", schemaTableName));
+        }
+    }
+
+    @Override
+    public void renameTable(ConnectorSession session, SchemaTableName from, SchemaTableName to)
+    {
+        try {
+            sessionCatalog.renameTable(convert(session), toIdentifier(from), toIdentifier(to));
+        }
+        catch (RESTException e) {
+            throw new TrinoException(ICEBERG_CATALOG_ERROR, e);
+        }
+    }
+
+    @Override
+    public Table loadTable(ConnectorSession session, SchemaTableName schemaTableName)
+    {
+        try {
+            return tableCache.computeIfAbsent(
+                    schemaTableName.toString(),
+                    key -> {
+                        BaseTable baseTable = (BaseTable) sessionCatalog.loadTable(convert(session), toIdentifier(schemaTableName));
+                        return new BaseTable(baseTable.operations(), IcebergUtil.quotedTableName(schemaTableName));
+                    });
+        }
+        catch (NoSuchTableException e) {
+            throw new TableNotFoundException(schemaTableName, e);
+        }
+        catch (Exception e) {
+            throw new TrinoException(ICEBERG_CATALOG_ERROR, e);
+        }
+    }
+
+    @Override
+    public void updateTableComment(ConnectorSession session, SchemaTableName schemaTableName, Optional<String> comment)
+    {
+        Table icebergTable = sessionCatalog.loadTable(convert(session), toIdentifier(schemaTableName));
+        if (comment.isEmpty()) {
+            icebergTable.updateProperties().remove(TABLE_COMMENT).commit();
+        }
+        else {
+            icebergTable.updateProperties().set(TABLE_COMMENT, comment.get()).commit();
+        }
+    }
+
+    @Override
+    public String defaultTableLocation(ConnectorSession session, SchemaTableName schemaTableName)
+    {
+        return null;
```

Please throw an exception if you don't expect this method is called. Note that this method wasn't nullable until this change.

---

good point, this should be fixed now